middle_end/flambda/types/structures/row_like.rec.ml

```diff
@@ -135,10 +151,20 @@ struct
     let ({ known_tags = known1; other_tags = other1; } : t) = t1 in
     let ({ known_tags = known2; other_tags = other2; } : t) = t2 in
     let env_extension = ref None in
+    let need_join =
```

Comment:
This isn't very easy to parse.
Apparently the only case where you don't need a join is when one side is a singleton (taking the `other_tags` field into account).
You could have written code that makes it more explicit.

Reply:
Would you prefer:
```OCaml
      | Bottom, _, Some _, _ -> false
      | _, Bottom, _, Some _ -> false
      | _ -> true
```
or
```OCaml
let do_not_need_join =
   other1 = Bottom && Tag.Map.is_singleton known1 ||
   other2 = Bottom && Tag.Map.is_singleton known2
```

Reply:
No polymorphic compare!

Reply:
The first one, with a comment explaining the reasoning if possible.

Reply:
Ok